src/pages/TraceDetail.tsx

```diff
@@ -226,6 +235,54 @@ function TraceDetail() {
 
         return nextSpans;
       });
+
+      // Mark the span as expanded
+      setExpandedSpans((prev) => new Set([...prev, spanId]));
+    });
+  };
+
+  const collapse = (spanId: string) => {
+    if (!result.isSuccess) {
+      return;
+    }
+
+    queryClient.setQueryData<Span[]>(queryKey, (oldData) => {
```

Comment:
I don't think I would manipulate the queryData.
I would update the `count` of `useVirtualizer` instead and deal with the hidden rows in the `rangeExtractor`.